client/my-sites/checkout/cart/cart-free-user-plan-upsell.tsx

```diff
@@ -17,186 +18,144 @@ import {
+interface CartFreeUserPlanUpsellProps {
 	cart: Pick< ResponseCart, 'products' >;
 	isCartPendingUpdate?: boolean;
 	addItemToCart: ( item: MinimalRequestCartProduct ) => void;
-	upsellPlan?: ReturnType< typeof getPlan >;
 }
 
-interface CartFreeUserPlanUpsellHocProps {
-	selectedSite?: SiteDetails | null;
-	hasPaidPlan?: boolean;
-	hasPlanInCart?: boolean;
-	isPlansListFetching?: boolean;
-	isRegisteringOrTransferringDomain?: boolean;
-	isSitePlansListFetching?: boolean;
-	planPrice: number | false | null | undefined;
-	showPlanUpsell?: boolean;
-	translate: ReturnType< typeof useTranslate >;
-	clickUpsellAddToCart: () => void;
-}
-
-class CartFreeUserPlanUpsell extends Component<
-	CartFreeUserPlanUpsellProps & CartFreeUserPlanUpsellHocProps
-> {
-	isLoading() {
-		const { isCartPendingUpdate: isLoadingCart } = this.props;
-		const isLoadingPlans = this.props.isPlansListFetching;
-		const isLoadingSitePlans = this.props.isSitePlansListFetching;
-		return isLoadingCart || isLoadingPlans || isLoadingSitePlans;
-	}
-
-	isRegistrationOrTransfer = ( item: ResponseCartProduct ) => {
-		return isDomainRegistration( item ) || isDomainTransfer( item );
-	};
-
-	getUpgradeText() {
-		const { cart, planPrice, translate, upsellPlan } = this.props;
-		const firstDomain = cart.products.find( this.isRegistrationOrTransfer );
-		const planName = upsellPlan?.getTitle() ?? '';
+const useUpsellPlanSlug = (): PlanSlug => {
+	return PLAN_PERSONAL;
+};
 
-		if ( planPrice && firstDomain && planPrice > firstDomain.cost ) {
-			const extraToPay = planPrice - firstDomain.cost;
-			return translate(
-				'Pay an {{strong}}extra %(extraToPay)s{{/strong}} for our %(planName)s plan, and get access to all its ' +
-					'features, plus the first year of your domain for free.',
-				{
-					args: {
-						extraToPay: formatCurrency( extraToPay, firstDomain.currency ),
-						planName,
-					},
-					components: {
-						strong: <strong />,
-					},
-				}
-			);
-		} else if ( planPrice && firstDomain && planPrice < firstDomain.cost ) {
-			const savings = firstDomain.cost - planPrice;
-			return translate(
-				'{{strong}}Save %(savings)s{{/strong}} when you purchase a WordPress.com %(planName)s plan ' +
-					'instead — your domain comes free for a year.',
-				{
-					args: {
-						planName,
-						savings: formatCurrency( savings, firstDomain.currency ),
-					},
-					components: {
-						strong: <strong />,
-					},
-				}
-			);
-		}
+const isDomainRegistrationOrTransfer = ( item: ResponseCartProduct ) => {
+	return isDomainRegistration( item ) || isDomainTransfer( item );
+};
 
+const UpgradeText = ( {
+	cart,
+	pricingMeta,
+}: {
+	cart: Pick< ResponseCart, 'products' >;
+	pricingMeta: Plans.PricingMetaForGridPlan;
+} ) => {
+	const translate = useTranslate();
+	const upsellPlanSlug = useUpsellPlanSlug();
+	const upsellPlan = getPlan( upsellPlanSlug );
+	const firstDomain = cart?.products?.find( isDomainRegistrationOrTransfer );
```

Comment:
Super minor but based on the types you don't need to use optional chaining here. A cart object is a required prop and it always has a `products` array.

```suggestion
	const firstDomain = cart.products.find( isDomainRegistrationOrTransfer );
```